File: ai/entity/aplist/apiListEntity.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class ApiList(Base):
    __tablename__ = "api_list"
    
    key = Column(Integer, primary_key=True, autoincrement=True)  # API 목록 키 (PK, AUTO_INCREMENT)
    id = Column(String(30), unique=True, nullable=False)         # API 목록 ID (프로젝트별 유니크)
    name = Column(String(20), nullable=False)                    # API 목록 명 (NOT NULL, 최대 20자)
    url = Column(String(50), nullable=False)                     # API 목록 URL (NOT NULL, 최대 50자)
    path = Column(String(100), nullable=False)                   # API 목록 경로 (NOT NULL, 최대 100자)
    method = Column(String(30), nullable=False)                  # API 목록 메서드 (NOT NULL, 최대 30자)
    description = Column(Text, nullable=True)                    # API 목록 설명 (TEXT)
    projectKey = Column(Integer, ForeignKey("project.key"), nullable=False)  # 프로젝트 키 (N:1)
    requestKey = Column(Integer, ForeignKey("request.key"), nullable=False)  # 요구사항 키 (N:1)
    
    # 관계 설정
    project = relationship("Project", back_populates="apiLists")
    request = relationship("Request", back_populates="apiLists")
    mappings = relationship("Mapping", back_populates="apiList", cascade="all, delete-orphan")
    
    def __init__(self, id: str, name: str, url: str, path: str, method: str, projectKey: int, requestKey: int, description: str = None):
        self.id = id
        self.name = name
        self.url = url
        self.path = path
        self.method = method
        self.projectKey = projectKey
        self.requestKey = requestKey
        self.description = description

File: backend/scenario/src/main/java/com/sk/skala/axcalibur/scenario/feature/apilist/entity/ApiListEntity.java
package com.sk.skala.axcalibur.scenario.feature.apilist.entity;

import com.sk.skala.axcalibur.scenario.global.entity.BaseTimeEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "api_list")
public class ApiListEntity extends BaseTimeEntity {
     
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "`key`")
    private Integer key;          // API 목록 키 (PK, AUTO_INCREMENT)

    @Column(name = "id", nullable = false, length = 30, unique = true)
    private String id;            // API 목록 ID (프로젝트별 유니크)

    @Column(name = "name", nullable = false, length = 20)
    private String name;                 // API 목록 명 (NOT NULL, 최대 20자)

    @Column(name = "uri", nullable = false, length = 50)
    private String uri;                  // API 목록 URL (NOT NULL, 최대 50자)

    @Column(name = "path", nullable = false, length = 100)
    private String path;                 // API 목록 경로 (NOT NULL, 최대 100자)

    @Column(name = "method", nullable = false, length = 30)
    private String method;               // API 목록 메서드 (NOT NULL, 최대 30자)

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;          // API 목록 설명 (TEXT)

    @Column(name = "project_key", nullable = false)
    private Integer projectKey;             // 프로젝트 키

    @Column(name = "request_key", nullable = false)
    private Integer requestKey;             // 요청 키
}